### DISCUSSION TITLE
How to catch abort errors in JavaScript?

### DISCUSSION THREAD

**Discussion by holdenmatt** [perspective-dev/perspective#1359] [Q&A] [answered]
## Support Question

How can I catch a Perspective error from JavaScript?

For example, if I pass a bad config to `table.view(config)` I see an uncaught error like this:
![image](https://user-images.githubusercontent.com/764380/111089348-0d4bfa80-84f1-11eb-8709-dcb699f47562.png)

But wrapping the offending call in a try/catch has no effect.

I didn't find anything about error handling in the documentation. This looks like what I want, but seems to only apply to Python/C++, not JavaScript:
https://github.com/finos/perspective/pull/777

What do you suggest? Thanks!

**Comment by texodus** [comment, marked as answer]
Due to the design of these "proxy" objects to their real-equivalents which live in the WebWorker, catching constructor errors is not possible up to version `0.6.2`.  On the `master` branch and thus in upcoming `0.7.0` release, the `view()` and `table()` factory methods are `async` and will throw constructor errors when invoked within an `async` context.  See #1289 for details.

**Reply by holdenmatt** [reply]
This sounds great, thanks for the reply!
